examples/htool_update_failure_reasons.h

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef THIRD_PARTY_LIBHOTH_LIBHOTH_EXAMPLES_HTOOL_UPDATE_FAILURE_REASONS_H_
+#define THIRD_PARTY_LIBHOTH_LIBHOTH_EXAMPLES_HTOOL_UPDATE_FAILURE_REASONS_H_
+
+#include <stdint.h>
+
+enum firmware_update_failure_reason {
+  FIRMWARE_UPDATE_SUCCESS = 0,  // EC_SUCCESS
+  FIRMWARE_UPDATE_NO_HEADER_FOUND = 1,
+  FIRMWARE_UPDATE_INVALID_HEADER_SIZE = 2,
+  FIRMWARE_UPDATE_INVALID_DESCRIPTOR = 3,
+  FIRMWARE_UPDATE_DELIVERY_MECHANISM_MISMATCH = 4,
+  FIRMWARE_UPDATE_INVALID_REGION = 5,
+  FIRMWARE_UPDATE_VERIFY_BAD_HEADER = 6,
+  FIRMWARE_UPDATE_VERIFY_HASH_IMAGE_FAILED = 7,
+  FIRMWARE_UPDATE_VERIFY_HASH_FUSE_MAP_FAILED = 8,
+  FIRMWARE_UPDATE_VERIFY_HASH_INFO_MAP_FAILED = 9,
+  FIRMWARE_UPDATE_VERIFY_SIGNATURE_FAILED = 10,
+  FIRMWARE_UPDATE_HASH_IMAGE_FIPS_FAILED = 11,
+  FIRMWARE_UPDATE_VERIFY_FIPS_FAILED = 12,
+  FIRMWARE_UPDATE_EXTERNAL_AB_HEADER_MISMATCH = 13,
+  FIRMWARE_UPDATE_VERSIONS_EQUAL = 14,
+  FIRMWARE_UPDATE_FIRST_VERSION_NEWER = 15,
+  FIRMWARE_UPDATE_MAUV_UPDATE_NOT_ALLOWED = 16,
+  FIRMWARE_UPDATE_EVEN_ODD_ROLLBACK_NOT_ALLOWED = 17,
+  FIRMWARE_UPDATE_EVEN_ODD_ROLLBACK_PAYLOAD_TOO_OLD = 18,
+  FIRMWARE_UPDATE_MIRROR_VERIFY_FAILED = 19,
+  FIRMWARE_UPDATE_MIRROR_RW_FAILED = 20,
+  FIRMWARE_UPDATE_MIRROR_RO_FAILED = 21,
+  FIRMWARE_UPDATE_VERSION_MATCHES_DENYLIST = 22,
+  FIRMWARE_UPDATE_INVALID_DESCRIPTOR_VERSION = 23,
+  FIRMWARE_UPDATE_INVALID_RW_KEY_TRANSITION = 24,
+  FIRMWARE_UPDATE_ERROR_MAX = 25,
+};
+
+// List of common failure codes for payload update.
+enum payload_update_failure_reason {
+  PAYLOAD_UPDATE_SUCCESS = 0,  // EC_SUCCESS
+  PAYLOAD_UPDATE_VALIDATE_RUNTIME_FAILURE =
+      1,  // VALIDATION_ERROR_RUNTIME_FAILURE
```

Comment:
nit: These comments are referring to the hoth firmware implementation, which isn't visible to htool anyhow; ok to remove them.

Reply:
Do you mean to remove all the comments?

Reply:
Removed all comments